src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertyAccessorSymbol.cs

```diff
@@ -181,6 +181,11 @@ internal override bool IsExpressionBodied
                     _name = overriddenMethod.Name;
                 }
             }
+
+            if (this.ReturnsByRefReadonly)
+            {
+                this.DeclaringCompilation.EnsureIsReadOnlyAttributeExists(this);
```

Comment:
>this.DeclaringCompilation.EnsureIsReadOnlyAttributeExists(this); [](start = 16, length = 64)

Shouldn't be done eagerly in the constructor, I think it would be better to bundle this check with some completion part, perhaps the one that binds return type.